src/settings.rs

```diff
@@ -0,0 +1,487 @@
+// Copyright 2025 Adobe. All rights reserved.
+// This file is licensed to you under the Apache License,
+// Version 2.0 (http://www.apache.org/licenses/LICENSE-2.0)
+// or the MIT license (http://opensource.org/licenses/MIT),
+// at your option.
+
+// Unless required by applicable law or agreed to in writing,
+// this software is distributed on an "AS IS" BASIS, WITHOUT
+// WARRANTIES OR REPRESENTATIONS OF ANY KIND, either express or
+// implied. See the LICENSE-MIT and LICENSE-APACHE files for the
+// specific language governing permissions and limitations under
+// each license.
+
+use c2pa::settings::Settings;
+use neon::prelude::*;
+use std::sync::{OnceLock, RwLock};
+
+use crate::runtime::reload_runtime;
+
+// Global JSON snapshot used to initialize each Tokio worker thread
+static GLOBAL_SETTINGS_TOML: OnceLock<RwLock<Option<String>>> = OnceLock::new();
+
+fn global_toml() -> &'static RwLock<Option<String>> {
+    GLOBAL_SETTINGS_TOML.get_or_init(|| RwLock::new(None))
+}
+
+fn set_global_settings_toml(toml: Option<String>) {
+    *global_toml().write().unwrap() = toml;
+}
+
+pub(crate) fn get_global_settings_toml() -> Option<String> {
+    global_toml().read().unwrap().clone()
+}
+
+/// Parse a JSON string (argument 0) into c2pa-rs Settings and apply them globally.
+/// Returns undefined; the JSON snapshot is stored globally for new worker threads.
+pub fn load_settings(mut cx: FunctionContext) -> JsResult<JsUndefined> {
+    let json_arg = cx.argument::<JsString>(0)?;
+    let json_string = json_arg.value(&mut cx);
+
+    // Apply settings globally using deprecated Settings::from_string with JSON format
```

Comment:
We'll need to undeprecate it, or switch to the new API that will support JSON. Whatever comes first.